File: src/Entity/Catalogue.php
<?php

namespace App\Entity;

use App\Repository\CatalogueRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=CatalogueRepository::class)
 * 
 */
class Catalogue
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     *  @Groups("data")
     */
    private $id;

    /**
     * 
     * @ORM\Column(type="string", length=255)
     *  @Groups("data")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     *
     *  @Groups("data")
     */
    private $description;

    /**
     * @ORM\OneToMany(targetEntity=Categorie::class, mappedBy="catalogue",cascade={"all"},orphanRemoval=true)
     * 
     */
    private $categories;

    public function __construct()
    {
        $this->categories = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    /**
     * @return Collection|Categorie[]
     */
    public function getCategories(): Collection
    {
        return $this->categories;
    }

    public function addCategory(Categorie $category): self
    {
        if (!$this->categories->contains($category)) {
            $this->categories[] = $category;
            $category->setCatalogue($this);
        }

        return $this;
    }

    public function removeCategory(Categorie $category): self
    {
        if ($this->categories->removeElement($category)) {
            // set the owning side to null (unless already changed)
            if ($category->getCatalogue() === $this) {
                $category->setCatalogue(null);
            }
        }

        return $this;
    }
    public function __toString()
    {
        return (string) $this->getNom();
    }
}

File: src/Entity/Catalogue.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entity;

/**
 *
 * @author WIKI
 */
public class Catalogue {
    private int id;
    private String nom;
    private String description;
    private Categorie categories;

    public Catalogue() {
    }

    public Catalogue(int id, String nom, String description, Categorie categories) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.categories = categories;
    }

     public Catalogue( String nom, String description, Categorie categories) {
       
        this.nom = nom;
        this.description = description;
        this.categories = categories;
    }

    public Catalogue(int id, String nom, String description) {
        this.id = id;
        this.nom = nom;
        this.description = description;
    }

    public Catalogue(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Categorie getCategories() {
        return categories;
    }

    public void setCategories(Categorie categories) {
        this.categories = categories;
    }

    @Override
    public String toString() {
        return "Catalogue{" + "id=" + id + '}';
    }


    
    
    
}
